tracing/span.go

```diff
@@ -306,11 +330,92 @@ func (s *Span) CreateLocalChild(name string, component string) *Span {
 // Parent id will be inherited from the span id,
 // and span id will be randomly generated.
 // Trace id, sampled, and flags will be copied over.
-func (s *Span) CreateLocalChildForContext(ctx context.Context, name string, component string) (context.Context, *Span) {
-	child := s.CreateLocalChild(name, component)
+func CreateLocalChildForContext(ctx context.Context, span Span, name, component string) (context.Context, *LocalSpan) {
+	child := CreateLocalChild(span, name, component)
 	return context.WithValue(ctx, childSpanKey, child), child
 }
 
+// ChildAndThriftContext creates both a client child span and a context can be
+// used by the thrift client code.
+//
+// A thrift client call would look like:
+//
+//     clientCtx, span := parentSpan.ChildAndThriftContext(ctx, "myCall")
+//     result, err := client.MyCall(clientCtx, arg1, arg2)
+//     span.End(ctx, err)
+func ChildAndThriftContext(ctx context.Context, span Span, name string) (context.Context, *ClientSpan) {
+	ctx, child := CreateClientChildForContext(ctx, span, name)
+	ctx = CreateThriftContextFromSpan(ctx, child)
+	return ctx, child
+}
+
+func initChildSpan(parent Span, child Span) {
+	for _, hook := range parent.Hooks() {
+		if err := hook.OnCreateChild(child); err != nil {
+			parent.LogError("OnCreateChild hook error: ", err)
+		}
+	}
```

Comment:
Should we call the hooks before or after we called `child.Start`? This is an example of things should be explained in the interface comments.

Reply:
After, because `OnCreateChild` can/is often used to register Hooks on the child span.

Reply:
But you called the hooks before you called `StartSpan` :)